### ISSUE TITLE
NotContains operator and empty array as a value - SQL error

### ISSUE THREAD

**Issue by DPHonys** [cube-js/cube#6583]
**Describe the bug**
When using the [notContains](https://cube.dev/docs/query-format#filters-operators-not-contains) operator in a filter with an empty array as a value Cube responds with an error "Error: syntax error at or near \"OR\"". Even though when using the [contains](https://cube.dev/docs/query-format#filters-operators-contains) operator with an empty array as a value Cube responds normally with data in response. My suspicion is a bug with SQL generation.

**To Reproduce**

1. Sent a request with a query (that uses contains operator) to the Cube API (I've used Postman)
  ```javascript
  {
    "measures": [
      "TaskCube.count"
    ],
    "filters": [
      {
        "member": "TaskCube.type",
        "operator": "contains",
        "values": []
      }
    ]
  }
  ```
  
  - Cube will generate SQL like
  ```bash
  2023-05-11 17:49:12 --
  2023-05-11 17:49:12   SELECT
  2023-05-11 17:49:12       count(*) "task_cube__count"
  2023-05-11 17:49:12     FROM
  2023-05-11 17:49:12       public.task AS "task_cube" 
  2023-05-11 17:49:12 --
  ```
  
  - Cube will respond with normal response with data
  ```javascript
  {
      "query": {
          "measures": [
              "TaskCube.count"
          ],
          "filters": [
              {
                  "operator": "contains",
                  "values": [],
                  "member": "TaskCube.type"
              }
          ],
          "timezone": "UTC",
          "order": [],
          "dimensions": [],
          "timeDimensions": [],
          "limit": 10000
      },
      "data": [
          {
              "TaskCube.count": "71706"
          }
      ],
  ...
  ```

2. Sent a request with a query (that uses notContains operator) to the Cube API (I've used Postman)
  ```javascript
  {
    "measures": [
      "TaskCube.count"
    ],
    "filters": [
      {
        "member": "TaskCube.type",
        "operator": "notContains",
        "values": []
      }
    ]
  }
  ```
  
  - Cube will generate SQL like
  ```bash
  2023-05-11 17:51:52 --
  2023-05-11 17:51:52   SELECT
  2023-05-11 17:51:52       count(*) "task_cube__count"
  2023-05-11 17:51:52     FROM
  2023-05-11 17:51:52       public.task AS "task_cube"  WHERE ( OR "task_cube".type IS NULL)
  2023-05-11 17:51:52 --
  ```
  
  - Cube will respond with SQL error
  ```javascript
  {
      "error": "Error: syntax error at or near \"OR\"",
      "requestId": "1ec01d6f-4041-473b-a19c-0b751e100872-span-1"
  }
  ```

**Expected behavior**
I would expect either the same response as with the operator "contains" or an error with a message something like "Expected one parameter but nothing found" as it is if you use an empty array with the operator "equals".

**Minimally reproducible Cube Schema**
```javascript
cube(`TaskCube`, {
  sql: `SELECT * FROM public.task`,

  measures: {
    count: {
      type: `count`,
    },
  },

  dimensions: {
    type: {
      sql: `type`,
      type: `string`,
    },
  },

  dataSource: `default`,
});

```

**Version:**
v0.33.5

**Comment by pacofvf** [comment]
@DPHonys That's an invalid query. All filter elements should contain a value. The "bug" is not the incorrect SQL, the "bug" here is that Cube didn't catch this syntax error earlier. Both of your queries with `contains` and `notContains` are invalid.

**Comment by DPHonys** [comment]
Ok, so it's not a "SQL bug" per se... but there is still inconsistency in responses... As for my Expected behavior, that means that Cube should respond with an "Expected one parameter but nothing found" error (if you've sent an empty array as values) for all operators except for `isSet` and `isNotSet` but it does not... so it's still a bug that is connected to as you've said, "Cube didn't catch this syntax error earlier".

**Comment by DPHonys** [comment]
Seems like it was fixed in version [0.33.12](https://github.com/cube-js/cube/compare/v0.33.11...v0.33.12). Cube now responds with
```json
{
    "error": "Values required for filter: {\"member\":\"TaskCube.type\",\"operator\":\"notContains\",\"values\":[]}",
    "stack": "Error: Values required for filter: {\"member\":\"TaskCube.type\",\"operator\":\"notContains\",\"values\":[]}\n    at /cube/node_modules/@cubejs-backend/api-gateway/src/query.js:142:13\n    at Array.map (<anonymous>)\n    at normalizeQueryFilters (/cube/node_modules/@cubejs-backend/api-gateway/src/query.js:122:10)\n    at normalizeQuery (/cube/node_modules/@cubejs-backend/api-gateway/src/query.js:209:14)\n    at /cube/node_modules/@cubejs-backend/api-gateway/src/gateway.ts:1113:49\n    at Array.map (<anonymous>)\n    at ApiGateway.getNormalizedQueries (/cube/node_modules/@cubejs-backend/api-gateway/src/gateway.ts:1111:15)\n    at ApiGateway.load (/cube/node_modules/@cubejs-backend/api-gateway/src/gateway.ts:1501:20)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at /cube/node_modules/@cubejs-backend/api-gateway/src/gateway.ts:246:7",
    "requestId": "2d768b3b-4990-4b5f-aa50-a2874ef30554-span-1"
}
```

Would be nice if @pacofvf could confirm. :)

**Comment by paveltiunov** [comment]
@DPHonys Yep. It was fixed in case you've used `queryRewrite`.
